### ISSUE TITLE
max-len sometimes fails to report errors when `ignoreComments` is true

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#6564]
**What version of ESLint are you using?** 2.13.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```json
{
  "rules": {
    "max-len": [2, 20, {"ignoreComments": true}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
// This commented line has the same length as the line below
var foo = 'this line is longer than the maximum line length'
```

**What did you expect to happen?**

I expected ESLint to report an error on the second line, since that line is longer than 20 characters.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint did not report any errors.

This bug only seems to occur when there is a long commented line somewhere in the file, and there is a non-commented line of exactly the same length later in the file.

**Comment by platinumazure** [comment]
Definitely seems like a bug. Thanks for the report!

**Comment by not-an-aardvark** [comment]
I think I have a fix for this; I'm reading the docs at the moment to figure out how to test it/make a PR correctly.
